web/src/main/java/org/cbioportal/web/GenesetHierarchyController.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2016 The Hyve B.V.
+ * This code is licensed under the GNU Affero General Public License (AGPL),
+ * version 3, or (at your option) any later version.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.web;
+
+import java.util.List;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+
+import org.cbioportal.model.GenesetHierarchyInfo;
+import org.cbioportal.service.GenesetHierarchyService;
+import org.cbioportal.service.exception.GeneticProfileNotFoundException;
+import org.cbioportal.web.config.annotation.InternalApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+@InternalApi
+@RestController
+@Validated
+@Api(tags = "Gene set hierarchy", description = " ")
+public class GenesetHierarchyController {
+
+    @Autowired
+    private GenesetHierarchyService genesetHierarchyService;
+
+    @RequestMapping(value = "/genesets/hierarchy/fetch", method = RequestMethod.POST,
```

Comment:
Can you change the path to `/geneset-hierarchy/fetch`, because we you say `/genesets/`, you are either getting a list of genesets, or a specific geneset resource, so having something other than a geneset id after `/genesets/` wouldn't make sense. `/geneset-hierarchy` looks like a different resource than `/genesets`.